package dsm;

enum MessageTypes {  
  MTYPE_PUT_REQUEST = 1;  
  MTYPE_GET_REQUEST = 2;
  MTYPE_GET_RESPONSE = 3;   
    
  MTYPE_WORKER_SHUTDOWN = 4;
  MTYPE_REGISTER_WORKER = 5;    
  
  MTYPE_WORKER_FLUSH = 6;
  MTYPE_WORKER_FLUSH_DONE = 7;
  
  MTYPE_SHARD_ASSIGNMENT = 8;
  MTYPE_SHARD_ASSIGNMENT_DONE = 9;    

  MTYPE_SYNC_REPLY = 10;  

  MTYPE_WORKER_APPLY = 11;
  MTYPE_WORKER_APPLY_DONE = 12;

};

message EmptyMessage {}

message Arg {
  required bytes key = 1;
  required bytes value = 2;
}

message Args {
  repeated Arg param = 1;
}

enum CompressionFormat {
  NONE = 0;
  LZO = 1;
  ZLIB = 2;
};

message FileParams {
  repeated Arg attr = 1;
  optional int32 compression = 2 [default=0];
}

message ConfigData {
  required int32 num_workers = 1;
  required int32 worker_id = 2;
  required int32 master_id = 3;
}
